pilot/util/harvester.py

```diff
@@ -244,19 +259,34 @@ def publish_work_report(work_report=None, worker_attributes_file="worker_attribu
 
     :param work_report: work report dictionary.
     :param worker_attributes_file:
-    :return:
+    :raises FileHandlingFailure: in case of IOError.
+    :return: True or False
     """
 
     if work_report:
-        work_report['timestamp'] = time_stamp()
-        if "outputfiles" in work_report:
-            del(work_report["outputfiles"])
-        if "inputfiles" in work_report:
-            del (work_report["inputfiles"])
-        if "xml" in work_report:
-            del (work_report["xml"])
-        if write_json(worker_attributes_file, work_report):
-            logger.info("work report published: {0}".format(work_report))
+        try:
+            work_report['timestamp'] = time_stamp()
+            if "outputfiles" in work_report:
+                del(work_report["outputfiles"])
+            if "inputfiles" in work_report:
+                del (work_report["inputfiles"])
+            if "xml" in work_report:
+                del (work_report["xml"])
+            if write_json(worker_attributes_file, work_report):
+                logger.info("work report published: {0}".format(work_report))
+                return True
+            else:
+                logger.info("work report publish failed: {0}".format(work_report))
```

Comment:
Use logger.warning here

Reply:
(I'm ignorant to the difference between logger.warning/error btw. if it's a fatal error i think i normally use logger.fatal)

Reply:
done changed else from info to error